carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -1556,4 +1556,26 @@ namespace carma_wm
     }
   }
 
+  std::optional<lanelet::ConstLanelet> CARMAWorldModel::getLaneletOnShortestPath(const std::vector<lanelet::ConstLanelet>& lanelets_to_filter) const
+  {
+    if (!route_ || lanelets_to_filter.empty())
+    {
+      return std::nullopt;
+    }
+
+    std::optional<lanelet::ConstLanelet> lanelet_on_the_route = std::nullopt;
+    // pick a lanelet on the shortest path
+    for (const auto& llt : lanelets_to_filter)
+    {
+      auto shortest_path = route_->shortestPath();
+      if (std::find(shortest_path.begin(), shortest_path.end(), llt) != shortest_path.end())
+      {
+        lanelet_on_the_route = llt;
```

Comment:
Given the current implementation, we can just return this value from here

Reply:
rearranged